### PR TITLE
Feat: Tags Field

### LINKED ISSUE #347: 'Tags' field plugin
## Summary

Hi, first thank you for this pretty tool, I love it.
I had a trouble with Gatsby and group-list and I think would be great if there was a Field that could parse markdown list, in particular I have an MDX file with this header: 

```
title: Title
description: Description
tags:
  - tag1
  - tag2
  - tag3
```

I've tried to parse tags with group-list in this way:

```
fields: [
    {
      label: 'Tags',
      name: 'rawFrontmatter.tags',
      component: 'group-list',
      itemProps: (item) => ({
        label: item,
      }),
      defaultItem: 'new-tag',
      fields: [
        { component: 'text' }
      ],
    },
  ...
```

But seems this is not the proper way to use group-list, I've also tried to add a `name` option but nothing.

I used to do that with NetlifyCMS with this configuration: 

```
fields:
      - { name: tags, label: Tags, widget: list }
```

### Basic example

I think that something like this could be much more easy to use:

```
fields: [
    {
      label: 'Tags',
      name: 'rawFrontmatter.tags',
      component: 'list',
      fields: [
        { name: 'tag', component: 'text' }
      ],
    },
  ...
```

### Motivation

This tool is great I want use it and I would see more people use it.

### PR DESCRIPTION
**Pull request by spbyrne**
Thanks to @DaniTulp we have a simple tags field, [see original PR here](https://github.com/tinacms/tinacms/pull/1146).

I've tweaked the styling, swapped the trash icon for an 'x' icon (just feels more appropriate) and I've added 'enter' to the list of keys that trigger a tag gets added when you're in the text field.

Screen Shot 2020-05-19 at 4 38 57 PM

Closes #347 

---
### Todo
- [x] Docs PR Ready

### CODE REVIEW

File: packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    text-overflow: ellipsis;
+    overflow: hidden;
+  }
+`
+export const TagsFieldPlugin = {
+  name: 'tags',
+  Component: TagsField,
+  validate(value: any, values: any, meta: any, field: any) {
+    if (field.required && !value) return 'Required'
+  },
```
Comment (ncphillips) [inline packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx:137]:
```suggestion
```

File: packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    <>
+      <BaseTextField
+        value={value}
+        onChange={event => setValue(event.target.value)}
+        placeholder={field.placeholder}
+        onKeyPress={event => {
+          if (event.key === ' ' || event.key === 'Enter') {
+            event.preventDefault()
+            addTag(value)
+            return
```
Comment (ncphillips) [inline packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx:52]:
```suggestion
```

File: packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  const [value, setValue] = React.useState<string>('')
+  const addTag = React.useCallback(
+    (tag: string) => {
+      if (form.getFieldState(field.name)?.value?.includes(tag)) {
+        return
+      }
+      form.mutators.insert(field.name, 0, tag)
+      setValue('')
+    },
+    [form, field]
```
Comment (ncphillips) [inline packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx:39]:
```suggestion
    [form, field.name]
```

**Comment by kendallstrautman** [comment]
Would be great to get some rudimentary documentation on this.

**Comment by kendallstrautman** [comment]
[Some docs](https://github.com/tinacms/tinacms.org/pull/424)

File: packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx (hunk 3 above)
Comment (spbyrne) [reply packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx]: Good catch 👍

File: packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx (hunk 2 above)
Comment (spbyrne) [reply packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx]: Removed

File: packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx (hunk 1 above)
Comment (spbyrne) [reply packages/@tinacms/fields/src/plugins/TagsFieldPlugin.tsx]: Good call, removed.

**Pull request merged**